### PR TITLE
include: clock: stm32f4 high speed clock definition

### PR DESCRIPTION
**Pull request by FRASTM**
Add the HSI and HSE clock source for the stm32f4 serie

Fixes https://github.com/zephyrproject-rtos/zephyr/discussions/80465

### CODE REVIEW

**Comment by FRASTM** [comment]
rebase on 9f93dede369c365c68b5618844e3d31e35327bc2
add sample for mco on some nucleo_f4 boards

File: samples/boards/st/mco/boards/nucleo_f429zi.overlay [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/* See reference manual (RM0090):
+ *   0b010: HSE clock selected
+ */
+#define MCO_SEL_HSE 2
+
+/* See reference manual (RM0090):
+ *   0b100: MCO divided by 2
+ */
+#define MCO_PRE_DIV_2 1
```
Comment (joerchan) [inline samples/boards/st/mco/boards/nucleo_f429zi.overlay:14]:
```suggestion
#define MCO_PRE_DIV_2 4
```
According to RM0090.

File: samples/boards/st/mco/boards/nucleo_f410rb.overlay [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/* See reference manual (RM0401):
+ *   0b010: HSE clock selected
+ */
+#define MCO_SEL_HSE 2
+
+/* See reference manual (RM0401):
+ *   0b100: MCO divided by 2
+ */
+#define MCO_PRE_DIV_2 1
```
Comment (joerchan) [inline samples/boards/st/mco/boards/nucleo_f410rb.overlay:14]:
```suggestion
#define MCO_PRE_DIV_2 4
```
According to RM0401

**Review by erwango — CHANGES_REQUESTED** [review]
Please remove (sqaush) 2 last commits

File: samples/boards/st/mco/boards/nucleo_f410rb.overlay
Comment (erwango) [inline samples/boards/st/mco/boards/nucleo_f410rb.overlay:1]: Let's remove this overlay as it doesn't have any added value and that just more file to maintain.

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by mathieuchopstm — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Comment by kartben** [comment]
@nordic-krch PTAL - Thanks!

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**